src/Microsoft.DotNet.Files/ContentFiles.cs

```diff
@@ -19,45 +20,68 @@ public ContentFiles(ProjectContext context)
         {
-            var sourceFiles = _context
-                .ProjectFile
-                .Files
-                .GetContentFiles();
+            if (includeEntries == null)
+            {
+                var sourceFiles = _context
+                    .ProjectFile
+                    .Files
+                    .GetContentFiles();
 
-            var sourceDirectory = _context.ProjectDirectory;
+                var sourceDirectory = _context.ProjectDirectory;
 
-            if (sourceFiles == null)
-            {
-                throw new ArgumentNullException(nameof(sourceFiles));
-            }
+                if (sourceFiles == null)
+                {
+                    throw new ArgumentNullException(nameof(sourceFiles));
+                }
 
-            sourceDirectory = EnsureTrailingSlash(sourceDirectory);
-            targetDirectory = EnsureTrailingSlash(targetDirectory);
+                sourceDirectory = EnsureTrailingSlash(sourceDirectory);
+                targetDirectory = EnsureTrailingSlash(targetDirectory);
 
-            var pathMap = sourceFiles
-                .ToDictionary(s => s,
-                    s => Path.Combine(targetDirectory,
-                        PathUtility.GetRelativePathIgnoringDirectoryTraversals(sourceDirectory, s)));
+                var pathMap = sourceFiles
+                    .ToDictionary(s => s,
+                        s => Path.Combine(targetDirectory,
+                            PathUtility.GetRelativePathIgnoringDirectoryTraversals(sourceDirectory, s)));
 
-            foreach (var targetDir in pathMap.Values
-                .Select(Path.GetDirectoryName)
-                .Distinct()
-                .Where(t => !Directory.Exists(t)))
-            {
-                Directory.CreateDirectory(targetDir);
-            }
+                foreach (var targetDir in pathMap.Values
+                    .Select(Path.GetDirectoryName)
+                    .Distinct()
+                    .Where(t => !Directory.Exists(t)))
+                {
+                    Directory.CreateDirectory(targetDir);
+                }
 
-            foreach (var sourceFilePath in pathMap.Keys)
-            {
-                File.Copy(
-                    sourceFilePath,
-                    pathMap[sourceFilePath],
-                    overwrite: true);
+                foreach (var sourceFilePath in pathMap.Keys)
+                {
+                    File.Copy(
+                        sourceFilePath,
+                        pathMap[sourceFilePath],
+                        overwrite: true);
+                }
+
+                RemoveAttributeFromFiles(pathMap.Values, FileAttributes.ReadOnly);
             }
+            else
+            {
+                foreach (var targetDir in includeEntries
+                    .Select(f => Path.GetDirectoryName(f.TargetPath))
+                    .Distinct()
+                    .Where(t => !Directory.Exists(t)))
```

Comment:
be careful of using Linq here. the content folder maybe deep, the `Distinct` could be costy.